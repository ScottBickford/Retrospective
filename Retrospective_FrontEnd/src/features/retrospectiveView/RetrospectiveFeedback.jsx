import React, { Fragment } from 'react';
import RetrospectiveFeedbackAdd from './RetrospectiveFeedbackAdd';

export const RetrospectiveFeedback = ({ feedback, handleAddFeedback }) => {
  return (
    <div style={{ paddingTop: 20 }}>
      <hr />
      <h4>Feedback:</h4>
      <hr />
      {(!feedback || feedback.length) === 0 && (
        <div>
          There is currently no feedback
          <hr />
        </div>
      )}
      {feedback &&
        feedback.length > 0 &&
        feedback.map(f => {
          return (
            <Fragment>
              <div
                key={f.feedbackId}
                style={{
                  backgroundColor: 'rgb(240,240,240)',
                  display: 'inline-block',
                  width: '100%'
                }}
              >
                <div style={{ float: 'left', width: '150px' }}>
                  <div>
                    <u>Name:</u>
                  </div>
                  {f.name}
                </div>
                <div style={{ float: 'left', width: '300px' }}>
                  <div>
                    <u>Body:</u>
                  </div>
                  {f.body}
                </div>
                <div style={{ float: 'left', width: '150px' }}>
                  <div>
                    <u>Type:</u>
                  </div>
                  {f.feedbackType}
                </div>
              </div>
              <hr />
            </Fragment>
          );
        })}
      <RetrospectiveFeedbackAdd handleAddFeedback={handleAddFeedback} />
    </div>
  );
};
