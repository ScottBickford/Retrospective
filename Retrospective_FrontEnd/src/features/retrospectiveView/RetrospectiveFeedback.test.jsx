import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RetrospectiveFeedback } from './RetrospectiveFeedback';

const feedback = [{feedbackId: 1, name: 'Person 1', body: 'Test Body', feedbackType: 'Positive'}];
describe('RetrospectiveFeedback', () => {
  test('snapshot renders', () => {    
    const component = renderer.create(<RetrospectiveFeedback feedback={feedback} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RetrospectiveFeedback feedback={feedback} />);
    const buttonElement = getByText(/Person 1/i);
    expect(buttonElement).toBeInTheDocument();
  });
});