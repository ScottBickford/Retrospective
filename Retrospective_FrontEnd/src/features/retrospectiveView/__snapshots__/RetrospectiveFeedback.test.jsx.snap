// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveFeedback snapshot renders 1`] = `
<div
  style={
    Object {
      "paddingTop": 20,
    }
  }
>
  <hr />
  <h4>
    Feedback:
  </h4>
  <hr />
  <div
    style={
      Object {
        "backgroundColor": "rgb(240,240,240)",
        "display": "inline-block",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "float": "left",
          "width": "150px",
        }
      }
    >
      <div>
        <u>
          Name:
        </u>
      </div>
      Person 1
    </div>
    <div
      style={
        Object {
          "float": "left",
          "width": "300px",
        }
      }
    >
      <div>
        <u>
          Body:
        </u>
      </div>
      Test Body
    </div>
    <div
      style={
        Object {
          "float": "left",
          "width": "150px",
        }
      }
    >
      <div>
        <u>
          Type:
        </u>
      </div>
      Positive
    </div>
  </div>
  <hr />
  <h4>
    Add feeback
  </h4>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        Name of person:
      </div>
      <input
        maxLength="200"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      />
    </div>
    <div>
      <div>
        Body:
      </div>
      <input
        maxLength="1000"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      />
    </div>
    <div>
      <div>
        Feedback type:
      </div>
      <select
        maxLength="1000"
        onChange={[Function]}
        style={
          Object {
            "width": 200,
          }
        }
        type="text"
        value=""
      >
        <option
          value=""
        >
          
        </option>
        <option
          value="Positive"
        >
          Positive
        </option>
        <option
          value="Negative"
        >
          Negative
        </option>
        <option
          value="Idea"
        >
          Idea
        </option>
        <option
          value="Praise"
        >
          Praise
        </option>
      </select>
    </div>
    <button
      disabled={true}
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Submit
    </button>
  </form>
</div>
`;
